components/api-controller/pkg/controller/networking/v1/networkingv1.go

```diff
@@ -157,3 +190,35 @@ func gatewayResourceFrom(vscv *istioNetworkingApi.VirtualService) *kymaMeta.Gate
 		Version: vscv.ResourceVersion,
 	}
 }
+
+func (a *istioImpl) isHostnameAvailable(metaDto meta.Dto, hostname string, assignedVirtualServiceUID types.UID) (bool, error) {
+
+	nsList, err := a.kubernetesInterface.CoreV1().Namespaces().List(k8sMeta.ListOptions{})
+
+	if err != nil {
+		return false, err
+	}
+
+	for _, ns := range nsList.Items {
+
+		vsList, err := a.istioVirtualServiceInterface(ns.GetName()).List(k8sMeta.ListOptions{})
+
+		if err != nil {
+			return false, err
+		}
+
+		for _, vs := range vsList.Items {
+			for _, occupiedHostname := range vs.Spec.Hosts {
+				if occupiedHostname == hostname {
+					//if hostname is not used by virtualservice actually assigned to CR
```

Comment:
`					//if hostname is used by virtualservice actually assigned to CR
					if vs.UID == assignedVirtualServiceUID {
						return true, nil
					}
                                          return false, nil
`

Reply:
Done, as it is the same suggestion as in the previous comment.